assets/js/index.js

```diff
@@ -14,4 +14,15 @@ document.addEventListener('DOMContentLoaded', () => {
   });
 });
 
+document.addEventListener('DOMContentLoaded', () => {
+  const loaderContainer = document.querySelector('#loader-container');
+  const reportSubmitButton = document.querySelector('#report-submit');
+  if (loaderContainer) {
+    document.querySelector('#report-submit .govuk-button').addEventListener('click', () => {
+      loaderContainer.classList.add('spinner-loader');
+      reportSubmitButton.classList.add('visuallyhidden');
+    });
+  }
+});
+
```

Comment:
Good job for adding this. This is in my TODO list for the confirm pages for all 3 flows once basic implementation is completed and tested.